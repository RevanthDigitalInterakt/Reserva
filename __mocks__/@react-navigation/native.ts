const navigate = jest.fn();

const useNavigation = () => ({
  navigate,
});

export {
  useNavigation,
};
