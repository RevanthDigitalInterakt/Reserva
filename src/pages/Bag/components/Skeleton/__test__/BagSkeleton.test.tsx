import React from 'react';
import { render, screen } from '@testing-library/react-native';
import { ThemeProvider } from 'styled-components/native';
import { theme } from '@usereservaapp/reserva-ui';
import BagSkeleton from '..';

const TestingComponent = (
  <ThemeProvider theme={theme}>
    <BagSkeleton />
  </ThemeProvider>
);

describe('BagSkeleton', () => {
  beforeEach(() => {
    render(TestingComponent);
  });

  it('should render properly and match with snapshot', () => {
    expect(screen.getByTestId('com.usereserva:id/BagSkeleton')).toBeVisible();
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
