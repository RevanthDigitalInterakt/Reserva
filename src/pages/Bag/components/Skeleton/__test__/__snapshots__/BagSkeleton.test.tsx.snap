// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BagSkeleton should render properly and match with snapshot 1`] = `
<View
  style={
    Object {
      "opacity": 1,
    }
  }
  testID="com.usereserva:id/BagSkeleton"
>
  <View
    marginLeft="xxs"
    marginRight="xs"
    style={
      Array [
        Object {
          "marginLeft": 24,
          "marginRight": 32,
        },
      ]
    }
  >
    <View
      bg="neutroFrio1"
      height={35}
      marginTop="xxs"
      style={
        Array [
          Object {
            "backgroundColor": "#D8D9DA",
            "height": 35,
            "marginTop": 24,
            "width": 175,
          },
        ]
      }
      width={175}
    />
    <View
      flexDirection="row"
      marginTop="xxs"
      style={
        Array [
          Object {
            "flexDirection": "row",
            "marginTop": 24,
          },
        ]
      }
    >
      <View
        bg="neutroFrio1"
        height={148}
        marginRight="nano"
        style={
          Array [
            Object {
              "backgroundColor": "#D8D9DA",
              "height": 148,
              "marginRight": 8,
              "width": 99,
            },
          ]
        }
        width={99}
      />
      <View
        height={148}
        style={
          Array [
            Object {
              "height": 148,
              "width": 217,
            },
          ]
        }
        width={217}
      >
        <View
          bg="neutroFrio1"
          height={22}
          style={
            Array [
              Object {
                "backgroundColor": "#D8D9DA",
                "height": 22,
                "width": 217,
              },
            ]
          }
          width={217}
        />
        <View
          flexDirection="row"
          style={
            Array [
              Object {
                "flexDirection": "row",
                "width": "100%",
              },
            ]
          }
          width="100%"
        >
          <View
            bg="neutroFrio1"
            height={18}
            marginRight="nano"
            marginTop="nano"
            style={
              Array [
                Object {
                  "backgroundColor": "#D8D9DA",
                  "height": 18,
                  "marginRight": 8,
                  "marginTop": 8,
                  "width": 79,
                },
              ]
            }
            width={79}
          />
          <View
            bg="neutroFrio1"
            height={18}
            marginTop="nano"
            style={
              Array [
                Object {
                  "backgroundColor": "#D8D9DA",
                  "height": 18,
                  "marginTop": 8,
                  "width": 79,
                },
              ]
            }
            width={79}
          />
        </View>
        <View
          bg="neutroFrio1"
          height={18}
          marginTop="nano"
          style={
            Array [
              Object {
                "backgroundColor": "#D8D9DA",
                "height": 18,
                "marginTop": 8,
                "width": 167,
              },
            ]
          }
          width={167}
        />
        <View
          bg="neutroFrio1"
          height={18}
          marginTop="nano"
          style={
            Array [
              Object {
                "backgroundColor": "#D8D9DA",
                "height": 18,
                "marginTop": 8,
                "width": 221,
              },
            ]
          }
          width={221}
        />
        <View
          bg="neutroFrio1"
          height={22}
          marginTop="nano"
          style={
            Array [
              Object {
                "backgroundColor": "#D8D9DA",
                "height": 22,
                "marginTop": 8,
                "width": 115,
              },
            ]
          }
          width={115}
        />
      </View>
    </View>
    <View
      bg="neutroFrio1"
      height={2}
      marginBottom="xxs"
      marginTop="xxs"
      style={
        Array [
          Object {
            "backgroundColor": "#D8D9DA",
            "height": 2,
            "marginBottom": 24,
            "marginTop": 24,
            "width": "100%",
          },
        ]
      }
      width="100%"
    />
    <View
      flexDirection="row"
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
        ]
      }
    >
      <View
        bg="neutroFrio1"
        height={22}
        marginRight="xxxs"
        style={
          Array [
            Object {
              "backgroundColor": "#D8D9DA",
              "height": 22,
              "marginRight": 16,
              "width": 232,
            },
          ]
        }
        width={232}
      />
      <View
        bg="neutroFrio1"
        height={22}
        style={
          Array [
            Object {
              "backgroundColor": "#D8D9DA",
              "height": 22,
              "width": 77,
            },
          ]
        }
        width={77}
      />
    </View>
    <View
      bg="neutroFrio1"
      height={2}
      marginBottom="xxs"
      marginTop="xxs"
      style={
        Array [
          Object {
            "backgroundColor": "#D8D9DA",
            "height": 2,
            "marginBottom": 24,
            "marginTop": 24,
            "width": "100%",
          },
        ]
      }
      width="100%"
    />
    <View
      bg="neutroFrio1"
      height={22}
      style={
        Array [
          Object {
            "backgroundColor": "#D8D9DA",
            "height": 22,
            "width": 232,
          },
        ]
      }
      width={232}
    />
    <View
      bg="neutroFrio1"
      height={47}
      marginTop="xxs"
      style={
        Array [
          Object {
            "backgroundColor": "#D8D9DA",
            "height": 47,
            "marginTop": 24,
            "width": "100%",
          },
        ]
      }
      width="100%"
    />
    <View
      flexDirection="row"
      marginTop="xxxs"
      style={
        Array [
          Object {
            "flexDirection": "row",
            "marginTop": 16,
            "width": "100%",
          },
        ]
      }
      width="100%"
    >
      <View
        bg="neutroFrio1"
        height={50}
        marginRight="micro"
        style={
          Array [
            Object {
              "backgroundColor": "#D8D9DA",
              "height": 50,
              "marginRight": 12,
              "width": 200,
            },
          ]
        }
        width={200}
      />
      <View
        bg="neutroFrio1"
        height={50}
        style={
          Array [
            Object {
              "backgroundColor": "#D8D9DA",
              "height": 50,
              "width": 122,
            },
          ]
        }
        width={122}
      />
    </View>
    <View
      flexDirection="row"
      marginTop="xxxs"
      style={
        Array [
          Object {
            "flexDirection": "row",
            "marginTop": 16,
            "width": "100%",
          },
        ]
      }
      width="100%"
    >
      <View
        bg="neutroFrio1"
        height={50}
        marginRight="micro"
        style={
          Array [
            Object {
              "backgroundColor": "#D8D9DA",
              "height": 50,
              "marginRight": 12,
              "width": 200,
            },
          ]
        }
        width={200}
      />
      <View
        bg="neutroFrio1"
        height={50}
        style={
          Array [
            Object {
              "backgroundColor": "#D8D9DA",
              "height": 50,
              "width": 122,
            },
          ]
        }
        width={122}
      />
    </View>
    <View
      flexDirection="row"
      marginTop="xxxs"
      style={
        Array [
          Object {
            "flexDirection": "row",
            "marginTop": 16,
            "width": "100%",
          },
        ]
      }
      width="100%"
    >
      <View
        bg="neutroFrio1"
        height={50}
        marginRight="micro"
        style={
          Array [
            Object {
              "backgroundColor": "#D8D9DA",
              "height": 50,
              "marginRight": 12,
              "width": 200,
            },
          ]
        }
        width={200}
      />
      <View
        bg="neutroFrio1"
        height={50}
        style={
          Array [
            Object {
              "backgroundColor": "#D8D9DA",
              "height": 50,
              "width": 122,
            },
          ]
        }
        width={122}
      />
    </View>
  </View>
</View>
`;
